### PR TITLE
[Example]Add GC-MC

### PR DESCRIPTION
**Pull request by jennyzhang0215**
## Description
Implemented GC-MC using bipartite graph API

The results on ML-100K is 0.92 without mini-batch training.

To run the code, download the movielens datasets (https://grouplens.org/datasets/movielens/) and unzip the datasets in the data_set/ml-?/ directory.

Now, we cannot run ml-1m, because of the "out of memory" error. We will refine the API and code soon. 

Paper reference: https://arxiv.org/pdf/1706.02263.pdf

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

**Comment by jermainewang** [comment]
Cool stuff! Could you add a readme (you may refer to readme of the other examples). Also, it seems to depend on GluonNLP. Is the dataset also provided there?

File: examples/mxnet/gc-mc/model.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    def __getitem__(self, key):
+        return self._layers[self._key2idx[key]]
+
+    def __contains__(self, key):
+        return key in self._key2idx
+
+
+class MultiLinkGCNAggregator(Block):
+    def __init__(self, g, src_key, dst_key, units, in_units, num_links,
+                 dropout_rate=0.0, accum='stack', act=None, **kwargs):
```
Comment (zheng-da) [inline examples/mxnet/gc-mc/model.py]: please document these arguments.

File: examples/mxnet/gc-mc/model.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+        src_input = self.dropout(src_input)
+        dst_input = self.dropout(dst_input)
+        #print("self._src_key", self._src_key)
+        #print("self._dst_key", self._dst_key)
+        self.g[self._src_key].ndata['fea'] = src_input
+        self.g[self._dst_key].ndata['fea'] = dst_input
+        def message_func(edges):
+            #print("\n\n In the message function ...")
+            msg_dic = {}
+            for i in range(self._num_links):
```
Comment (zheng-da) [inline examples/mxnet/gc-mc/model.py]: I'm confused by this `_num_links`.

File: examples/mxnet/gc-mc/train.py [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+    possible_rating_values = dataset.possible_rating_values
+    nd_possible_rating_values = mx.nd.array(possible_rating_values, ctx=args.ctx, dtype=np.float32)
+    train_rating_pairs = mx.nd.array(dataset.train_rating_pairs, ctx=args.ctx, dtype=np.int64)
+    train_gt_ratings = mx.nd.array(dataset.train_rating_values, ctx=args.ctx, dtype=np.float32)
+    rating_mean = dataset.train_rating_values.mean()
+    rating_std = dataset.train_rating_values.std()
+
+    print("Start preparing graph ...")
+    uv_train_graph = dataset.uv_train_graph
+    vu_train_graph = dataset.vu_train_graph
```
Comment (zheng-da) [inline examples/mxnet/gc-mc/train.py]: should we remove the training edges from these two graphs? It seems we can use edge_subgraph to remove the training edges from the original graph.

**Comment by jermainewang** [comment]
Closed and moved to #860

**Pull request closed without merging**